lib/puppet/provider/postgresql_psql/ruby.rb

```diff
@@ -16,19 +16,21 @@ def run_sql_command(sql)
     command.push("-p", resource[:port]) if resource[:port]
     command.push("-t", "-c", '"' + sql.gsub('"', '\"') + '"')
 
+    environment = get_environment
+
     if resource[:cwd]
       Dir.chdir resource[:cwd] do
-        run_command(command, resource[:psql_user], resource[:psql_group])
+        run_command(command, resource[:psql_user], resource[:psql_group], environment)
       end
     else
-      run_command(command, resource[:psql_user], resource[:psql_group])
+      run_command(command, resource[:psql_user], resource[:psql_group], environment)
     end
   end
 
   private
 
   def get_environment
-    environment = {}
+    environment =  resource[:connect_settings] ? resource[:connect_settings] : Hash.new;
```

Comment:
This could also be simplified by:

`environment = resource[:connect_settings] || {}`